caffe2: Fix lint errors in native/quantized/TensorAdvancedIndexing

EricGriffith opened the pull request:
Summary: See title

Test Plan: Sandcastle

Differential Revision: D72273049

malfet commented on aten/src/ATen/native/quantized/TensorAdvancedIndexing.cpp:
```diff
@@ -182,9 +182,9 @@ Tensor& _index_put_impl_quantized_cuda_(Tensor & self, const torch::List<std::op
-    if (index.has_value()) {
-      at::assert_no_overlap(self, *index);
+  for (const auto index: indices) {
+    if (index.get().isTensor()) {
```
It's late, and I might not know how to read C++, but `std::optional<>` does not have a method `.get()` does it?

malfet commented on aten/src/ATen/native/quantized/TensorAdvancedIndexing.cpp:
```diff
@@ -182,9 +182,9 @@ Tensor& _index_put_impl_quantized_cuda_(Tensor & self, const torch::List<std::op
-      at::assert_no_overlap(self, *index);
+  for (const auto index: indices) {
+    if (index.get().isTensor()) {
+      at::assert_no_overlap(self, index.get().toTensor());
```
Why is it better than before?

malfet reviewed (changes requested):
I'm not sure where method `get()` and `isTensor()` are coming from

EricGriffith committed:
caffe2: Fix lint errors in native/quantized/TensorAdvancedIndexing.cpp (#150504)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/150504

See title

Test Plan: Sandcastle

Differential Revision: D72273049
